src/ast/ast_validations.c

```
@@ -1,2478 +1,168 @@
+	
+	const char *alias = cypher_ast_identifier_get_name(identifier);
+	if (!alias || strlen(alias) == 0) return 0;
+	
+	// Get the labels for this node
+	uint label_count = cypher_ast_node_pattern_nlabels(node_pattern);
+	
+	dictEntry *entry = dictFind(node_labels, (char *)alias);
+	
+	if (!entry) {
+		// First occurrence - store the labels for this alias
+		array labels = array_new(cypher_astnode_t *, label_count);
```

Cppcheck (2.21.0)

[error] Common realloc mistake

(memleakOnRealloc)